spec/operators/catch-spec.ts

```diff
@@ -242,4 +242,15 @@ describe('Observable.prototype.catch', () => {
       done();
     });
   });
+
+  it('should allow types with operators such as mergeMap to flow correctly', () => {
+    type(() => {
+      Observable.from([{ a: 1 }, { a: 2 }, { a: 3} ])
+          .mergeMap(value => {
+              return Observable.of([value]);
+          })
+          .mergeMap(x => x)
+          .subscribe(x => x.a);
+    });
+  });
```

Comment:
> before the changes this code will fail.

: Would you able to elaborate what failure you've experienced? as @blesh said, it seems bit odd since operator does not involves any path to `catch` at all.